starknet-foundry/crates/cheatable-starknet/src/rpc.rs

```diff
@@ -0,0 +1,499 @@
+                return Ok(());
+            }
+        }
+        self.syscall_handler.execute_next_syscall(vm, hint)
+    }
+
+    fn verify_syscall_ptr(&self, actual_ptr: Relocatable) -> SyscallResult<()> {
+        if actual_ptr != self.syscall_handler.syscall_ptr {
+            return Err(SyscallExecutionError::BadSyscallPointer {
+                expected_ptr: self.syscall_handler.syscall_ptr,
+                actual_ptr,
+            });
+        }
+
+        Ok(())
+    }
+}
+
+/// Executes a specific call to a contract entry point and returns its output.
+fn execute_entry_point_call_cairo1(
+    call: CallEntryPoint,
+    contract_class: &ContractClassV1,
+    state: &mut dyn State,
+    cheated_state: &CheatedState,
+    resources: &mut ExecutionResources,
+    context: &mut EntryPointExecutionContext,
+) -> EntryPointExecutionResult<CallInfo> {
+    let VmExecutionContext {
+        mut runner,
+        mut vm,
+        mut syscall_handler,
+        initial_syscall_ptr,
+        entry_point,
+        program_segment_size,
+    } = initialize_execution_context(call, contract_class, state, resources, context)?;
+
+    let args = prepare_call_arguments(
+        &syscall_handler.call,
+        &mut vm,
+        initial_syscall_ptr,
+        &mut syscall_handler.read_only_segments,
+        &entry_point,
+    )?;
+    let n_total_args = args.len();
+
+    // Snapshot the VM resources, in order to calculate the usage of this run at the end.
+    let previous_vm_resources = syscall_handler.resources.vm_resources.clone();
+
+    let mut syscall_hh = CheatableSyscallHandler {
+        syscall_handler,
+        cheated_state,
+    };
+
+    // Execute.
+    let run_resources = cheatable_run_entry_point(
+        &mut vm,
+        &mut runner,
+        &mut syscall_hh,
+        &entry_point,
+        &args,
+        program_segment_size,
+    )?;
+
+    let sys_h = syscall_hh.syscall_handler;
+    sys_h.context.vm_run_resources = run_resources;
+    let call_info = finalize_execution(vm, runner, sys_h, previous_vm_resources, n_total_args)?;
+    if call_info.execution.failed {
+        return Err(EntryPointExecutionError::ExecutionFailed {
+            error_data: call_info.execution.retdata.0,
+        });
+    }
+
+    Ok(call_info)
+}
+
+/// Runs the runner from the given PC.
+fn cheatable_run_entry_point(
+    vm: &mut VirtualMachine,
+    runner: &mut CairoRunner,
+    hint_processor: &mut dyn HintProcessor,
+    entry_point: &EntryPointV1,
+    args: &Args,
+    program_segment_size: usize,
+) -> Result<RunResources, VirtualMachineExecutionError> {
+    let verify_secure = false;
+    let args: Vec<&CairoArg> = args.iter().collect();
+
+    let mut run_resources = RunResources::default();
+    runner.run_from_entrypoint(
+        entry_point.pc(),
+        &args,
+        &mut run_resources,
+        verify_secure,
+        Some(program_segment_size),
+        vm,
+        hint_processor,
+    )?;
+
+    Ok(run_resources)
+}
```

Comment:
Was comparing with blockifier, why do we care only about run resources?

Reply:
What do you mean? I return them here for compatibility, since i can't pass mutable reference into that function

Reply:
In blockifier something else is returned I think, so I was wondering if we are not losing any data here